go/vt/topo/shard.go

```diff
@@ -648,47 +646,20 @@ func (ts *Server) GetTabletsByShardCell(ctx context.Context, keyspace, shard str
 	defer span.Finish()
 	var err error
 
-	if len(cells) == 0 {
-		cells, err = ts.GetCellInfoNames(ctx)
-		if err != nil {
-			return nil, err
-		}
-		if len(cells) == 0 { // Nothing to do
-			return nil, nil
-		}
+	// if we get a partial result, we keep going. It most likely means
+	// a cell is out of commission.
+	aliases, err := ts.FindAllTabletAliasesInShardByCell(ctx, keyspace, shard, cells)
+	if err != nil && !IsErrType(err, PartialResult) {
+		return nil, err
```

Comment:
The changes to this func make it correct, but slower. Previously the keyspace and shard were being ignored and we'd return all tablets. Where is this func being used, and what is the impact of fetching each tablet from topo versus (in the best case) a single topo call fetching all tablets?

Reply:
It is being used only in VTOrc to refresh the tablets in a specific keyspace shard.

Reply:
OK. Typically the number of tablets in a shard is small, so this seems correct.